compiler/src/lnntoamm/Fn.ts

```diff
@@ -0,0 +1,235 @@
+import { LPNode, NulLP } from '../lp';
+import Output from './Amm';
+import opcodes from './opcodes';
+import Scope from './Scope';
+import Stmt, { Assign, Dec, Emit, Exit, FnArg, MetaData, Ref } from './Statement';
+import Type, { FunctionType } from './Types';
+import { TODO } from './util';
+
+// the value is null if the type is to be inferred
+export type Args = {[name: string]: FnArg};
```

Comment:
Then why not `{[name: string]: FnArg | null}`

Reply:
this comment is slightly outdated - it used to be `export type Args = {[name: string]: Type | null};`, but I've since added the `FnArg` class and ensure that each instance's type isn't `null`. I'll delete this comment since it no longer applies